modules/REST/V2/Languages.php

```diff
@@ -0,0 +1,737 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST\V2;
+
+use PLL_Language;
+use PLL_Model;
+use stdClass;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Allows to access languages via the REST API.
+ *
+ * @since 3.7
+ *
+ * @phpstan-type Fields array{
+ *     code?: non-empty-string,
+ *     locale?: non-empty-string,
+ *     name?: non-empty-string,
+ *     direction?: 'ltr'|'rtl',
+ *     order?: int,
+ *     flag?: string,
+ *     set_default_cat?: bool
+ * }
+ */
+class Languages extends WP_REST_Controller {
+	/**
+	 * @var PLL_Model
+	 */
+	public $model;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param PLL_Model $model Polylang's model.
+	 */
+	public function __construct( PLL_Model $model ) {
+		$this->namespace = 'pll/v2';
+		$this->rest_base = 'languages';
+		$this->model     = $model;
+	}
+
+	/**
+	 * Registers the routes for languages.
+	 *
+	 * @since 3.7
+	 *
+	 * @return void
+	 */
+	public function register_routes(): void {
```

Comment:
I wonder if we should set `$override` parameter of `register_rest_route()` to `true`, so our code takes precedence. What do you think?

Reply:
We never used it so far. I'm not against but I'm not sure in which cases it could be useful though 🤔

Reply:
Done in [4d6e713](https://github.com/polylang/polylang/pull/1505/commits/4d6e7137232a89b55be4b13a7c9d4a4c300642d5).